lib/puppet/provider/rabbitmq_binding/rabbitmqadmin.rb

```diff
@@ -50,13 +50,19 @@ def self.instances
           arguments = '{}'
         end
         if (source_name != '')
-          binding = {
-            :destination_type => destination_type,
-            :routing_key      => routing_key,
-            :arguments        => JSON.parse(arguments),
-            :ensure           => :present,
-            :name             => "%s@%s@%s" % [source_name, destination_name, vhost],
-          }
+          if not binding[:name]
```

Comment:
The original code that was deleted creates a hash and assigns it to the variable 'binding'.  This new code assumes the variable that points to the hash already exists -- does it exist yet?  This is the same as saying `nil[:name]`, which causes an error.

Consider: `if not binding`